src/main/java/com/tma/demo/service/post/imp/PostServiceImp.java

```diff
@@ -67,24 +64,72 @@ public PostDto createPost(String content, MultipartFile[] mediaFiles) {
     }
 
+    @Override
+    public PostDto updatePost(String postId,
+                              MultipartFile[] files,
+                              String content,
+                              String[] deleteFiles) {
+
+        Post post = postRepository.findPostById(UUID.fromString(postId))
+                .orElseThrow(() -> new BaseException(ErrorCode.POST_NOT_FOUND));
+        post.setContent(content);
+        saveAllMediaFiles(files, post);
+        List<UUID> deletedFileIds = Arrays.stream(deleteFiles).map(UUID::fromString).toList();
+        deleteMedia(deletedFileIds, UUID.fromString(postId));
+        deleteMediaInCloud(deleteFiles, postId, FolderNameConstant.POST);
+        List<Media> mediaList = getMediaByPostId(UUID.fromString(postId));
+        PostDto parentPost = getParentPost(post.getParentPost());
+        return postMapper.from(post, mediaList, parentPost);
+    }
+
+
+    private PostDto getParentPost(Post post) {
+        if (post == null) {
+            return null;
+        }
+        if (post.getParentPost() == null) {
+            return postMapper.from(post, getMediaByPostId(post.getId()), null);
+        }
+        return getParentPost(post.getParentPost());
+    }
+
     //    =================================================================================================================
 //    MEDIA
     private List<Media> saveAllMediaFiles(MultipartFile[] mediaFiles, Post post) {
         List<Media> mediaList = new ArrayList<>();
         for (MultipartFile mediaFile : mediaFiles) {
-            String uuid = UUID.randomUUID().toString();
-            Map data = cloudinaryService.upload(mediaFile, FolderNameConstant.POST, post.getId() + "/" + uuid);
+
             Media media = Media.builder()
                     .isDelete(false)
-                    .mediaUrl(data.get(AttributeConstant.CLOUDINARY_URL).toString())
                     .mediaType(MediaType.IMAGE)
                     .post(post)
                     .build();
+            media = mediaRepository.saveAndFlush(media);
+            Map data = cloudinaryService.upload(mediaFile, FolderNameConstant.POST, post.getId() + "/" + media.getId());
+            media.setMediaUrl(data.get(AttributeConstant.CLOUDINARY_URL).toString());
             mediaList.add(mediaRepository.saveAndFlush(media));
         }
         return mediaList;
     }
 
+    private List<Media> getMediaByPostId(UUID postId) {
+        return mediaRepository.findAllByPostId(postId);
+    }
+
+    private void deleteMedia(List<UUID> deleteFiles, UUID postId) {
+        List<Media> mediaList = mediaRepository.findAllByIdsAndPostId(deleteFiles, postId);
+        System.out.println(mediaList);
+        mediaRepository.deleteAll(mediaList);
+    }
+
+    private void deleteMediaInCloud(String[] deleteFiles, String prefix, String folder) {
+        for (String deleteFile : deleteFiles) {
+            String publicId = folder + (prefix != null ? "/" + prefix : "") + "/" + deleteFile;
```

Comment:
Create constant EMPTY = "" and OLIDUS = "/" and use String.format() for folder + (prefix != null ? "/" + prefix : "") + "/" + deleteFile